src/components/Avatar/Avatar.tsx

```diff
@@ -53,7 +53,28 @@ export interface Props {
   variant?: 'icon' | 'initials' | 'image';
 }
 
-function getInitials(fromName: string): string {
+/**
+ * Algo for names with special characters in:
+ * - Split name parts by space
+ * - for a given part,
+ *  - if that part has emoji or extended characters, {Action?}
+ *  - if that part does not, then use the first ASCII character
+ * @param str
+ * @returns
+ */
+function produceAbbreviation(str: string): string {
+  // unicode character range for surrogate pairs (first and second halves)
+  if (str.match(/([\uD800-\uDFFF])*/g)?.filter((match) => !!match).length) {
```

Comment:
Could probably use use unicode emoji character class thing to match emoji:

```js
/\p{Emoji}/u.test(str)
```

which may be more readable